[colinac] iP

colinac opened the pull request:
### Bob

> Never leave till tomorrow that which you can do today.

Bob is a chatbot who is in need of a break. 
He serves you to the best of his abilities and can:

- [ ] List tasks
- [ ] Mark / Unmark them
- [ ] Create deadlines
- [ ] ~Do your tasks~ (unfortunately)
- [ ] Create events
- [ ] Create todos
- [ ] Delete tasks


To start using Bob, 
1. Git clone his repo [here](https://github.com/colinac/ip)
2. Open the project in Intellij
3. Run the program by typing ``./gradlew run`` in the terminal

Get Bob now and get your tasks done in no time! 🥳


_If you are a Java programmer, you can use it to practice Java too. Here's the main method:_
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

colinac committed:
Change greeting and bot name and remove traces of Duke from source code

colinac committed:
Add echo functionality

colinac committed:
Add ability to store text and display them back to user when requested

colinac committed:
Add ability to mark tasks as done.

colinac committed:
Add support for tracking to-dos, deadlines and events

colinac committed:
Add indent for task for task addition message

colinac committed:
Remove spaces which cause FC to fail

colinac committed:
Add test cases and update runtest.bat

colinac committed:
Add exception handling

colinac committed:
Add support for deleting tasks from the list, update tests and fix bugs

colinac committed:
Add ability to save tasks in the hard disk automatically whenever the task list changes

colinac committed:
Specify file paths in an OS-independent way

colinac committed:
Merge branch 'branch-level-7'

colinac committed:
Teach chatbot how to understand dates and times

colinac committed:
Merge branch 'branch-Level-8'

colinac committed:
Update unit tests

colinac committed:
Add more oop

colinac committed:
Merge branch 'branch-A-MoreOOP'

colinac committed:
Place classes in to Bob package

colinac committed:
Update script

colinac committed:
Update gitignore

colinac committed:
Merge branch 'add-gradle-support' of https://github.com/colinac/ip

colinac committed:
Correct package name

colinac committed:
Update build.gradle

colinac committed:
Merge branch 'branch-A-Gradle'

colinac committed:
Add unit tests for Event and Todo classes

colinac committed:
Merge branch 'branch-A-JUnit'

colinac committed:
Update build.gradle

colinac committed:
Merge branch 'branch-A-Jar'

colinac committed:
Add JavaDoc comments

colinac committed:
Add blank line

colinac committed:
Add find functionality

colinac committed:
Adhere to coding standards

colinac committed:
Merge branch 'branch-A-JavaDoc'

colinac committed:
Merge branch 'branch-A-CodingStandard'

colinac committed:
Merge branch 'branch-Level-9'

Chronoxy reviewed (commented):
Hello, I have read through your code. It seems like you did not violate any Naming Conventions Standards. 😄

BuffWuff1712 commented on src/main/java/bob/HelpCommand.java:
I like how you added a Help command to assist users who might be unfamiliar with what commands to use

BuffWuff1712 commented on src/main/java/bob/MatchListCommand.java:
is there supposed to be future functionality for MatchListCommand?

colinac replied:
No, nothing planned for it at the moment 😄

BuffWuff1712 reviewed (commented):
Overall, the code is well-organised with OOP principles adhered to. Great job👍

colinac committed:
Configure checkstyle

colinac committed:
Merge branch 'branch-A-Checkstyle'

colinac committed:
Update runtest.bat

colinac committed:
Implement gui

colinac committed:
Update help message

colinac committed:
Merge branch 'branch-Level-10'

colinac committed:
Add assertions to classes

Some assumptions in the code are only verified if a bug occurs.

This can cause one to be less certain of the robustness of his code,
even making debugging harder as he has to go through each assumption
and verify that they are not the cause of the bug.

Using assertions in the classes allows us to check if a certain
behaviors are valid at each point in the code.

By doing so, one can have more confidence in the code and narrow down
the cause of a new bug to areas where assertions have failed or
areas areas not sufficiently covered by assertions, reducing areas
in the code in which one has to search for the bug.

colinac committed:
Improve code quality

Code has certain flaws, with missing comments and lack of
prioritization of the happy path being examples of them.

Poor code quality makes it difficult to find bugs and correct them.
Lack of comments, for example, maintains the obscurity of the code,
increasing the difficulty of understanding why the code was written in
a certain way.

Make the happy path more prominent and describe the ErrorBob and
InvalidCommandException classes, along with the parseDatetime method.

Making the happy path more prominent reduces nesting while comments
on the specified classes and method reduce ambiguity about the
roles they perform.

colinac committed:
Merge pull request #1 from colinac/branch-A-Assertions

Add assertions to classes

colinac committed:
Merge branch 'master' into branch-A-CodeQuality

colinac committed:
Merge pull request #2 from colinac/branch-A-CodeQuality

Improve code quality

colinac committed:
Add minor enhancements to UI

colinac committed:
Merge branch 'branch-ui-enhancement'

colinac committed:
Add line breaks for exception triggered responses

colinac committed:
Add createdAt for tasks

colinac committed:
Move classes into relevant packages

colinac committed:
Merge branch 'master' into branch-BCD-Extension

colinac committed:
Add summary functionality

colinac committed:
Merge branch 'branch-BCD-Extension'

colinac committed:
Add JavaDoc comments for select classes

colinac committed:
Update unit tests

colinac committed:
Merge branch 'branch-update-junit'

colinac committed:
Correct help message and command parsing

colinac committed:
Merge branch 'master' into branch-A-UserGuide

colinac committed:
Update user guide

colinac committed:
Correct parts of the README

colinac committed:
Merge branch 'branch-A-UserGuide'

colinac committed:
Set title for GUI window

colinac committed:
Update image in README

colinac committed:
Edit Java version

colinac committed:
Update README.md

colinac committed:
Update README.md

colinac committed:
Update README.md
